src/main/webapp/app/complaints/form/complaints-form.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ComplaintService } from 'app/complaints/complaint.service';
+import { Result } from 'app/entities/result.model';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Complaint, ComplaintType } from 'app/entities/complaint.model';
+import { Exercise } from 'app/entities/exercise.model';
+import { onError } from 'app/shared/util/global.utils';
+import { AlertService } from 'app/core/util/alert.service';
+
+@Component({
+    selector: 'jhi-complaint-form',
+    templateUrl: './complaints-form.component.html',
+    styleUrls: ['complaints-form.component.scss'],
+})
+export class ComplaintsFormComponent implements OnInit {
+    @Input() exercise: Exercise;
+    @Input() resultId: number;
+    @Input() examId?: number;
+    @Input() allowedComplaints: number; // the number of complaints that a student can still submit in the course
+    maxComplaintsPerCourse: number;
+    @Input() complaintType: ComplaintType;
+    @Input() isCurrentUserSubmissionAuthor = false;
+    @Output() submit: EventEmitter<void> = new EventEmitter();
+    complaintText?: string;
+    ComplaintType = ComplaintType;
+
+    constructor(private complaintService: ComplaintService, private alertService: AlertService) {}
+
+    ngOnInit(): void {
+        if (this.exercise.course) {
+            this.maxComplaintsPerCourse = this.exercise.teamMode ? this.exercise.course.maxTeamComplaints! : this.exercise.course.maxComplaints!;
+        } else {
+            this.maxComplaintsPerCourse = 1;
```

Comment:
Could we add a `DEFAULT` / `CONSTANT` variable here?

Reply:
Do you mean initialising the attribute with `1` and leaving out the else clause then? If not, please specify more.

Reply:
either like you described or:
`this.maxComplaintsPerCourse = DEFAULT_MAX_COURSE_COMPLAINT;`
and specify this constant at the top of the file or somewhere else.

Reply:
Ah, I see. I'll do my version then since it's simpler. But good point with the default value.